burn-train/src/learner/builder.rs

```diff
@@ -210,8 +247,26 @@ where
         Optim::Record: 'static,
         LR::Record: 'static,
     {
-        self.init_logger();
-        let callback = Box::new(self.dashboard);
+        if self.log_to_file {
+            self.init_logger();
+        }
+        let renderer = self
+            .renderer
+            .unwrap_or_else(|| Box::new(CLIDashboardRenderer::new()));
+        let directory = &self.directory;
+        let logger_train = self.metric_logger_train.unwrap_or_else(|| {
+            Box::new(FileMetricLogger::new(format!("{directory}/train").as_str()))
+        });
+        let logger_valid = self.metric_logger_valid.unwrap_or_else(|| {
+            Box::new(FileMetricLogger::new(format!("{directory}/valid").as_str()))
+        });
+        let mut dashboard = Dashboard::new(renderer, logger_train, logger_valid);
+        dashboard.metrics_train = self.metrics_train;
+        dashboard.metrics_valid = self.metrics_valid;
+        dashboard.metrics_train_numeric = self.metrics_train_numeric;
+        dashboard.metrics_valid_numeric = self.metrics_valid_numeric;
```

Comment:
If we go for a metric struct, we could pass it as parameter in the `new` instead. It would also remove all methods on the dashboard to add metrics, since they should be passed in the `new` function from now on. Those functions could be moved to the newly created metric struct if we see fit, but making each field `pub(crate)` may also be the better option.